File: Python/3/4.py
# Required for math.sqrt() function
import math

# Get input from user
num = int(input("Enter a number: "))

# Initilize divisor as 2
divisor = 2

# Till divisor < square root of num
while divisor <= int(math.sqrt(num)):
    # If remainder is zero number is not prime
    if num % divisor == 0:
        print("Number is not prime")
        break
    
    # Increment divisor by 1
    divisor += 1

# If num % divisor is not zero number is prime
else:
    print("Number is prime")

File: C/3/4.c
#include <stdio.h>

int i, num, factors = 0;

int main()
{
    // Get input from user
    printf("Enter a number: ");
    scanf("%d", &num);

    // Calculate the number of factors
    for (i = 1; i <= num; i++)
    {
        if (num % i == 0)
        {
            factors++;
        }
    }

    // Print whether the number is prime or composite
    if (factors > 2)
    {
        printf("\n%d is composite\n", num);
    }
    else if (factors == 2)
    {
        printf("\n%d is prime\n", num);
    }
    else
    {
        printf("\n%d is neither prime nor composite\n", num);
    }

    return 0;
}
